chiselFrontend/src/main/scala/Chisel/internal/Builder.scala

```diff
@@ -120,4 +120,13 @@ private[Chisel] object Builder {
       Circuit(components.last.name, components)
     }
   }
+
+  def buildModule[T <: Module](f: => T): T = {
+    dynamicContextVar.withValue(Some(new DynamicContext)) {
+      errors.info("Elaborating design...")
+      val mod = f
+      mod.forceName(mod.name, globalNamespace)
+      mod
+    }
+  }
```

Comment:
So the naming here is a little weird, both `build` and `buildModule` do almost identical things, one just returns a Circuit while one returns the Module type itself. They should probably share most of their code, but also we should consider renaming them